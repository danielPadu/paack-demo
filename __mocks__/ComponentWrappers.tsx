import {NavigationContext} from '@react-navigation/native';
import React from 'react';
import {Provider} from 'react-redux/src';
import {PersistGate} from 'redux-persist/integration/react';
import {persistor} from '../appFiles/appStore/store';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import {defaultReducer} from '../appFiles/appStore/reducers/default';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
export const mockedStore = mockStore(defaultReducer);

export const navContext = {
  isFocused: () => true,
  // addListener returns an unscubscribe function.
  addListener: jest.fn(() => jest.fn()),
  navigate: Function.prototype,
  setParams: Function.prototype,
  dispatch: Function.prototype,
  getParam: (param: {deliveryId?: string} & any, defaultValue: any) => {
    return defaultValue;
  },
};

export const ComponentWrappers = (props: {
  children: React.ReactNode;
  redux_store: any;
}) => {
  const {children, redux_store} = props;

  return (
    <>
      <Provider store={redux_store ?? mockedStore}>
        <PersistGate loading={null} persistor={persistor}>
          <NavigationContext.Provider value={navContext as any}>
            {children}
          </NavigationContext.Provider>
        </PersistGate>
      </Provider>
    </>
  );
};
