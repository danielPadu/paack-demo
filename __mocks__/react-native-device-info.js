export default {
  getModel: jest.fn(),
  getVersion: jest.fn(),
  getBuildNumber: jest.fn(),
  getBundleId: jest.fn(),
  hasNotch: jest.fn(),
};
