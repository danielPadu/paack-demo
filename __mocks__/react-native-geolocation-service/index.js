export default {
  addListener: jest.fn(),
  getCurrentPosition: () => ({
    coords: {
      accuracy: 0,
      altitude: 0,
      altitudeAccuracy: 0,
      heading: 0,
      latitude: 0,
      longitude: 0,
      speed: 0,
    },
  }),
  removeListeners: jest.fn(),
  requestAuthorization: jest.fn(),
  setConfiguration: jest.fn(),
  startObserving: jest.fn(),
  stopObserving: jest.fn(),
};
