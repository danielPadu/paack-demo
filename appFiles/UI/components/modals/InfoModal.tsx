import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {ScrollView} from 'react-native-gesture-handler';
import Modal from 'react-native-modal';
import {Card, Colors} from 'react-native-ui-lib';
import {useDispatch, useSelector} from 'react-redux';
import {infoModalClose} from '../../../appStore/actions/modalActions';
import {screenHeight, screenWidth} from '../../utils';

const InfoModal = () => {
  const {
    infoModal: {isOpen, options},
  } = useSelector((state: any) => state.modalReducer);
  const dispatch = useDispatch();
  const {title, description} = options;
  const {modalStyle, cardStyle, shadow, titleStyle, descriptionStyle} = styles;
  const modalClosing = () => {
    if (isOpen) {
      dispatch(infoModalClose());
    }
  };

  return isOpen ? (
    <Modal
      animationIn="fadeIn"
      animationOut="fadeOut"
      animationInTiming={400}
      propagateSwipe={true}
      isVisible={isOpen}
      backdropColor={Colors.primaryBackdropColor}
      onBackdropPress={() => modalClosing()}
      style={modalStyle}
      deviceWidth={screenWidth}
      deviceHeight={screenHeight}>
      <Card
        row={false}
        style={[cardStyle, shadow]}
        enableShadow={true}
        useNative
        activeOpacity={1}>
        <Text style={titleStyle}>{title}</Text>
        <ScrollView showsVerticalScrollIndicator={false}>
          <Text style={descriptionStyle}>{description}</Text>
        </ScrollView>
      </Card>
    </Modal>
  ) : null;
};
const styles = StyleSheet.create({
  modalStyle: {
    flex: 1,
    margin: 0,
    marginBottom: 0,
    justifyContent: 'center',
    alignItems: 'center',
  },
  cardStyle: {
    flex: 0,
    flexShrink: 1,
    marginHorizontal: '15%',
    marginVertical: '10%',
    padding: 10,
    flexDirection: 'column',
    alignItems: 'center',
    height: 200,
  },
  shadow: {
    borderRadius: 20,
    shadowColor: Colors.primaryBackdropColor,
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.39,
    shadowRadius: 8.3,

    elevation: 3,
  },
  titleStyle: {
    textAlign: 'center',
    fontWeight: '900',
    fontSize: 24,
    color: Colors.blue1,
    paddingVertical: 20,
  },
  descriptionStyle: {
    fontWeight: '400',
    fontSize: 16,
    color: Colors.blue1,
    paddingVertical: 20,
    textAlign: 'justify',
  },
});
export default InfoModal;
