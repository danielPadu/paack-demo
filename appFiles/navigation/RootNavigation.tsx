// RootNavigation.js
import * as React from 'react';
import {createNavigationContainerRef} from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
 
export const navigationRef = createNavigationContainerRef<RootStackParamList>();
export type RootStackParamList = {
  IntroScreen: {itemId?: string};
  DeliveriesScreen: {itemId?: string};
  DeliveryDetailsScreen: {deliveryId?: string; itemId?: string};
};
export type DeliveryScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'DeliveriesScreen'
>;
export type DeliveryDetailsScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'DeliveryDetailsScreen'
>;
export type IntroScreenProps = NativeStackScreenProps<
  RootStackParamList,
  'IntroScreen'
>;



export const navigate = (name: keyof RootStackParamList, params?: unknown) => {
  if (navigationRef.isReady()) {
    // Perform navigation if the app has mounted
    console.log('navigationRef navigating to : ', name, );
    navigationRef.navigate(name as never, params as never);
  } else {
    // You can decide what to do if the app hasn't mounted
    // You can ignore this, or add these actions to a queue you can call later
    console.log('CAN NOT NAVIGATE ');
  }
};
 