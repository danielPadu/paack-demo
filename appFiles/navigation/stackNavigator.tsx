import React, {useEffect} from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {IntroScreen, DeliveriesScreen, DeliveryDetailsScreen} from '../screens';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {navigationRef, isReadyRef} from './RootNavigation';
//import MembershipActiveScreen from '../screens/components/MembershipActiveScreen';
import {useDispatch} from 'react-redux';
import {setAppActiveScreen} from '../appStore/actions/appActions';
export type RootStackParamList = {
  IntroScreen: undefined;
  DeliveriesScreen: undefined;
  DeliveryDetailsScreen: {deliveryId: string};
};
const Stack = createStackNavigator();
const defaultOptions = {
  animationEnabled: false,
  swipeEnabled: false,
};
const STACK_PROPS = (props: {
  initialRouteName: string;
  children: React.ReactNode | React.ReactNode[];
}) => (
  <Stack.Navigator
    screenOptions={{
      headerShown: true,
      title: '',
      headerStyle: {height: 30},
    }}
    initialRouteName={
      props.initialRouteName ? props.initialRouteName : undefined
    }>
    {props.children}
  </Stack.Navigator>
);

// Gets the current screen from navigation state
const getActiveRouteName = state => {
  const route = state?.routes[state?.index];
  if (route?.state) {
    // Dive into nested navigators
    return getActiveRouteName(route?.state);
  }
  return route;
};

const AppContainer = () => {
  const previousActiveRouteRef = React.useRef();
  const activeRouteRef = React.useRef();
  const dispatch = useDispatch();

  useEffect(() => {
    //  const { removeOnNotificationOpened, removeOnNotification, removeOnTokenRefresh } = onlySetTheListners(); // set listners for FCM and detach them when app closes
    const state = navigationRef.current?.getRootState();
    activeRouteRef.current = getActiveRouteName(state); // Save the initial route name
    return () => {
      console.log('AppContainer unmounted');
      if (isReadyRef?.current) {
        (isReadyRef.current as React.MutableRefObject<boolean>).current = false;
      }
    };
  }, []);

  const theme = {
    //like this
    colors: {
      // background: "transparent",
      opacity: 0.75,
    },
  };
  return (
    <SafeAreaProvider>
      <NavigationContainer
        // theme={theme}
        // initialState={initialState}
        ref={navigationRef}
        // removeClippedSubviews={false}
        onStateChange={state => {
          const previousActiveRoute = activeRouteRef.current;
          const currentRoute = getActiveRouteName(state);
          if (previousActiveRoute !== activeRouteRef) {
            // console.log('NavigationContainer previousActive: -> ', previousActiveRoute);
            console.log('NavigationContainer active: -> ', currentRoute);
            dispatch(setAppActiveScreen(currentRoute.name));
          }
          // Save the current route name for later comparision
          activeRouteRef.current = currentRoute;
          previousActiveRouteRef.current = previousActiveRoute;
        }}
        onReady={() => {
          if (isReadyRef?.current) {
            (isReadyRef?.current as React.MutableRefObject<boolean>).current =
              true;
          }
        }}>
        <STACK_PROPS initialRouteName="IntroScreen">
          <Stack.Screen
            name="IntroScreen"
            component={IntroScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'Intro'}}
          />
          <Stack.Screen
            name="DeliveriesScreen"
            component={DeliveriesScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'DeliveriesScreen'}}
          />
          <Stack.Screen
            name="DeliveryDetailsScreen"
            component={DeliveryDetailsScreen}
            options={() => ({
              backBehavior: 'order',
              ...defaultOptions,
            })}
            initialParams={{itemId: 'DeliveryDetailsScreen'}}
          />
        </STACK_PROPS>
      </NavigationContainer>
    </SafeAreaProvider>
  );
};

export default AppContainer;
