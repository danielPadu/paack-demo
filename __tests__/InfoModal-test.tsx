import React from 'react';
import renderer, {
  act,
  ReactTestRenderer,
  ReactTestRendererJSON,
} from 'react-test-renderer';
import InfoModal from '../appFiles//UI/components/modals/InfoModal';
import {ComponentWrappers, mockedStore} from '../__mocks__/ComponentWrappers';
import {navigate} from '../appFiles/navigation/RootNavigation';
import {waitFor} from 'react-native-testing-library';
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest
    .fn()
    .mockReturnValueOnce({
      infoModal: {
        isOpen: true,
        options: {title: 'Title', description: 'description'},
      },
    })
    .mockReturnValueOnce({
      infoModal: {
        isOpen: false,
        options: {title: 'Title', description: 'description'},
      },
    }),
  useDispatch: () => jest.fn(),
}));
let store;
let component:
  | ReactTestRenderer
  | ReactTestRenderer[]
  | ReactTestRendererJSON
  | ReactTestRendererJSON[]
  | null;
describe('checking LoadingModal component', () => {
  test('InfoModal renders correctly', async () => {
    await act(async () => {
      await beforeEach(() => {
        store = mockedStore;

        component = renderer
          .create(
            <ComponentWrappers redux_store={store}>
              <InfoModal />
            </ComponentWrappers>,
          )
          .toJSON();
        expect(component).toMatchSnapshot();
        expect(navigate('Deliveries')).toBeFalsy();
      });
    });
  });
  test('InfoModal inner functionallity tests', async () => {
    await act(async () => {
      await beforeEach(async () => {
        store = mockedStore;

        component = renderer
          .create(
            <ComponentWrappers redux_store={store}>
              <InfoModal />
            </ComponentWrappers>,
          )
          .toJSON();
        const {findByProps} = component;
        const indicator = findByProps('size');
        expect(component.props.isOpen).toEqual(true);
        expect(component.props.modalClosing()).toEqual(true);
        await waitFor(() => expect(indicator).toBeTruthy());
        await waitFor(() => expect(component).toHaveProperty('containerStyle'));
      });
      await afterEach(() => {
        // cleanup on exiting
        component?.remove?.();
        component = null;
      });
    });
  });
});
