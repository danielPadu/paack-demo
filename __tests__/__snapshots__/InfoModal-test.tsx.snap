// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checking LoadingModal component InfoModal inner functionallity tests 1`] = `
<Modal
  animationType="none"
  deviceHeight={1334}
  deviceWidth={750}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={false}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={1334}
    deviceWidth={750}
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "marginBottom": 0,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      accessibilityState={
        Object {
          "selected": undefined,
        }
      }
      activeOpacity={1}
      delayPressIn={10}
      enableShadow={true}
      row={false}
      style={
        Array [
          undefined,
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "borderRadius": 12,
              "height": undefined,
              "overflow": "visible",
              "width": undefined,
            },
            Object {
              "borderRadius": 12,
            },
            Object {
              "elevation": 2,
            },
            Object {
              "shadowColor": "#A6ACB1",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 12,
            },
            Object {
              "backgroundColor": "#FFFFFF",
            },
            undefined,
            Array [
              Object {
                "alignItems": "center",
                "flex": 0,
                "flexDirection": "column",
                "flexShrink": 1,
                "height": 200,
                "marginHorizontal": "15%",
                "marginVertical": "10%",
                "padding": 10,
              },
              Object {
                "borderRadius": 20,
                "elevation": 3,
                "shadowColor": undefined,
                "shadowOffset": Object {
                  "height": 6,
                  "width": 0,
                },
                "shadowOpacity": 0.39,
                "shadowRadius": 8.3,
              },
            ],
          ],
        ]
      }
      useNative={true}
    >
      <Text
        style={
          Object {
            "color": "#05214D",
            "fontSize": 24,
            "fontWeight": "900",
            "paddingVertical": 20,
            "textAlign": "center",
          }
        }
      >
        Title
      </Text>
      <RCTScrollView
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        showsVerticalScrollIndicator={false}
      >
        <View>
          <Text
            style={
              Object {
                "color": "#05214D",
                "fontSize": 16,
                "fontWeight": "400",
                "paddingVertical": 20,
                "textAlign": "justify",
              }
            }
          >
            description
          </Text>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
